minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/Mixins/MixinLargeInventorySlots.java

```diff
@@ -0,0 +1,52 @@
+package com.microsoft.Malmo.Mixins;
+
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.inventory.Container;
+import net.minecraft.inventory.ContainerPlayer;
+import net.minecraft.inventory.Slot;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.util.TreeSet;
+
+@Mixin(ContainerPlayer.class)
+public abstract class MixinLargeInventorySlots extends Container {
+    @Inject(method="<init>*", at=@At("RETURN"))
+    public void constructorTail(CallbackInfo ci) {
+        InventoryPlayer inventory = null;
+        TreeSet<Integer> missingIndices = new TreeSet<Integer>();
+        for (Slot slot : this.inventorySlots) {
+            if (slot.inventory instanceof InventoryPlayer) {
+                if (inventory == null) {
+                    inventory = (InventoryPlayer) slot.inventory;
+                    for (int i = 0; i < inventory.mainInventory.size(); i++) {
+                        missingIndices.add(i);
+                    }
+                }
+                if (slot.slotNumber >= 36) {
+                    // These are armor and other equipment slots, we shift their indices by 360 - 36 since we've
+                    // expanded the previous 36-long inventory to 360.
+                    slot.slotNumber += inventory.mainInventory.size() - 36;
+                }
+                else {
+                    missingIndices.remove(slot.slotNumber);
+                }
+            }
+        }
+        if (inventory == null) {
+            // TODO: should raise an error?
+            return;
+        }
+        while (missingIndices.size() > 0) {
+            int slot = missingIndices.first();
+            // Note: the first "slot" is the actual index, the other two are what the x and y position should be in the
```

Comment:
Maybe we could be more clever with  slot % int(sqt(slot)), slot / int(sqt(slot))? It is possible to open the inventory as a human and would be nice to be able to see what was in it

Reply:
My guess is that there are so many slots that no matter what it will look like a giant mess in the UI. I think we can wait until we cross that bridge for now.